### PR TITLE
initial implementation of darkmode

### PR DESCRIPTION
**Pull request by feydkin**
Regarding issue #18

- mixin solution to detect prefers-color-scheme
- modified chart to pick up theme (or force a theme) - currently requires a page reload

To Do's
1. completely convert existing styling method (variable file)
2. convert and test mobile styling
3. implement a light/dark toggle override button

### CODE REVIEW

File: src/app/components/pool-size-chart/pool-size-chart.component.ts [hunk 1]
```diff
@@ -26,14 +27,23 @@ export class PoolSizeChartComponent implements OnInit {
   readonly chartBaseUnit = 1024 ** 4; // TB
 
   get palette(): PoolSizeChartPalette {
+    if ( this.theme == null ) {
```
Comment (aervin) [inline src/app/components/pool-size-chart/pool-size-chart.component.ts:30]: small nit: === is safer

File: src/app/services/api/color-scheme.service.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    private colorScheme: string;
+    // Define prefix for clearer and more readable class names in scss files
+    private colorSchemePrefix = 'color-scheme-';
+
+    constructor(rendererFactory: RendererFactory2) {
+        // Create new renderer from renderFactory, to make it possible to use renderer2 in a service
+        this.renderer = rendererFactory.createRenderer(null, null);
+    }
+
+    _detectPrefersColorScheme() {
```
Comment (aervin) [inline src/app/services/api/color-scheme.service.ts:18]: i like the leading _. could mark these private as well

**Review by aervin — COMMENTED** [review]
Looks good! Only some minor notes from me. No mandatory changes.

File: src/app/components/header/header.component.scss [hunk 3]
```diff
@@ -93,15 +95,18 @@ header {
...
     &:focus {
-      border-color: #ffffff61;
+      @include theme('border-color', 'accent-color-dark');
     }
   }
 
+/*****************************************************************************
+*   MOBILE
+******************************************************************************/
+
```
Comment (undsoft) [inline src/app/components/header/header.component.scss:109]: Not everything down is mobile, `.action-button` is not.
Also I think `mobile-menu-toggle` is self descriptive enough and we don't need a separator.

File: src/app/services/api/color-scheme.service.ts (hunk 2 above)
Comment (undsoft) [reply src/app/services/api/color-scheme.service.ts:18]: I'd rather we not have leading underscores. It's a remnant from javascript days to indicate that method is private. In Typescript you can just use `private` keyword for this.

It's not even recommended in javascript: https://github.com/airbnb/javascript#naming--leading-underscore

File: src/app/services/api/color-scheme.service.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import {Injectable, Renderer2, RendererFactory2} from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ColorSchemeService {
+
+    private renderer: Renderer2;
+    private colorScheme: string;
```
Comment (undsoft) [inline src/app/services/api/color-scheme.service.ts:9]: How about introducing a enum instead of string here?

**Review by undsoft — CHANGES_REQUESTED** [review]
I like the approach.
The charts look good, switching between themes works well.

1. I am not a fan of white header. How about trying another darker shade of black with some border if necessary?
2. Lines in tables and around some buttons are too strong now in both light and dark theme.
3. I liked the gray github contribution prompt more.
4. Need to somehow separate footer from articles on dark theme now.
5. Remember to relint.

Thanks

File: src/app/components/header/header.component.scss (hunk 3 above)
Comment (feydkin) [reply src/app/components/header/header.component.scss:109]: Can you double check this? Maybe I am misreading but it looks like everything below the separator is part of either mobile-menu-toggle and mobile-menu classes. There is also an action-button subclass under the header class above this.

**Review by undsoft — APPROVED** [review]
It looks good, thanks!

One minor thing is the new stats bar. It looks good on darkmode, but on light mode it's white, which doesn't work for front page - too contrasty, since everything else is dark.

**Comment by aervin** [comment]
Thanks @feydkin ! Let's get this onto develop so we can get more eyes on it.

**Pull request merged**